### PR TITLE
Attempted to complete the Homework: ticket service hierarchy

### PR DESCRIPTION
**Pull request by rektasan**

### CODE REVIEW

File: src/main/java/Main/Main.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package Main;
+
+public class Main {
+
+  public static void main(String[] args) {
+
```
Comment (AtticSiol) [inline src/main/java/Main/Main.java]: add some engine (new object) or move this method in TicketService class

File: src/main/java/Model/Ticket.java [hunk 2]
```diff
@@ -19,7 +25,8 @@ public class Ticket {
   private StadiumSectors stadiumSector;
   private BigDecimal ticketPrice;
 
-  public Ticket(String concertHall, int eventCode, long time, boolean isPromo, StadiumSectors stadiumSector, double maxBackpackWeight,
+
+  public Ticket(int classId, String concertHall, int eventCode, long time, boolean isPromo, StadiumSectors stadiumSector, double maxBackpackWeight,
       BigDecimal ticketPrice) {
 
     setConcertHall(concertHall);
```
Comment (AtticSiol) [inline src/main/java/Model/Ticket.java]: remove redundant setters

File: src/main/java/Model/Ticket.java [hunk 3]
```diff
@@ -149,10 +160,25 @@ public LocalDateTime getCREATION_TIME() {
...
+  }
+
+
+  @Override
+  public void print() {
+    System.out.println("Ticket's print() method : " + this);
+  }
+
   @Override
   public String toString() {
```
Comment (AtticSiol) [inline src/main/java/Model/Ticket.java]: check the order of fields and methods

File: src/main/java/Main/Main.java (hunk 1 above)
Comment (rektasan) [reply src/main/java/Main/Main.java]: Practiced with object instantiation and polymorphism in main().

File: src/main/java/Model/Ticket.java (hunk 2 above)
Comment (rektasan) [reply src/main/java/Model/Ticket.java]: Removed boiler-plate code, using lombok. Moved validation to a separate class.

File: src/main/java/Model/Ticket.java (hunk 3 above)
Comment (rektasan) [reply src/main/java/Model/Ticket.java]: Checked the order of fields and methods.

File: src/main/java/com/jfb/tickets/model/ticket/Ticket.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  }
+
+  @Override
+  public int getClassId() {
+    return this.classId;
+  }
+
+  @Override
+  public void setClassId(int id) {
```
Comment (AtticSiol) [inline src/main/java/com/jfb/tickets/model/ticket/Ticket.java]: if you override this method from interface, it is better to generate value for id already there, for example in default method.

File: src/main/java/com/jfb/tickets/util/interfaces/Identifiable.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.jfb.tickets.util.interfaces;
+
+public interface Identifiable {
```
Comment (AtticSiol) [inline src/main/java/com/jfb/tickets/util/interfaces/Identifiable.java:5]: each class has to have id (even ticket service)

File: src/main/java/com/jfb/tickets/model/ticket/Ticket.java (hunk 4 above)
Comment (rektasan) [reply src/main/java/com/jfb/tickets/model/ticket/Ticket.java]: Made sure id is generated inside Identifiable interface.

File: src/main/java/com/jfb/tickets/util/interfaces/Identifiable.java (hunk 5 above)
Comment (rektasan) [reply src/main/java/com/jfb/tickets/util/interfaces/Identifiable.java:5]: Made sure each class has an id field. Good catch!

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**